collectors/bzimport/convertors.py

```diff
@@ -548,6 +548,45 @@ def flags(self):
 
         return [flag for flag in self.flaw_bug["flags"] if isinstance(flag, dict)]
 
+    @staticmethod
+    def get_major_incident_state(flags):
+        """
+        Gets a Major Incident state from hightouch and hightouch-lite flags.
+        """
+        # Default values
+        hightouch = ""
+        hightouch_lite = ""
+        major_incident_state = Flaw.FlawMajorIncident.INVALID
+
+        # Sets values from Bugzilla
+        for flag in flags:
+            if flag["name"] == "hightouch":
+                hightouch = flag["status"]
+            if flag["name"] == "hightouch-lite":
+                hightouch_lite = flag["status"]
+
+        valid_pairs = [
+            [("", ""), Flaw.FlawMajorIncident.NOVALUE],
+            [("?", "?"), Flaw.FlawMajorIncident.REQUESTED],
+            [("?", ""), Flaw.FlawMajorIncident.REQUESTED],
+            [("", "?"), Flaw.FlawMajorIncident.REQUESTED],
+            [("-", "-"), Flaw.FlawMajorIncident.REJECTED],
+            [("-", ""), Flaw.FlawMajorIncident.REJECTED],
+            [("", "-"), Flaw.FlawMajorIncident.REJECTED],
+            [("+", ""), Flaw.FlawMajorIncident.APPROVED],
+            [("+", "-"), Flaw.FlawMajorIncident.APPROVED],
+            [("", "+"), Flaw.FlawMajorIncident.CISA_APPROVED],
+            [("-", "+"), Flaw.FlawMajorIncident.CISA_APPROVED],
+        ]
```

Comment:
I think `dict` mapping would be a more suitable in this case:

```python
valid_pais = {
    ("", "") : Flaw.FlawMajorIncident.NOVALUE,
    ...
}
```

Then you can simply check it for expected value and avoid iterating through the whole thing.